examples/linux/io_epoll_test.cpp

```diff
@@ -155,6 +172,43 @@ int main() {
   auto buffer = std::vector<char>{};
   buffer.resize(1);
   try {
+#if !UNIFEX_NO_COROUTINES
+    auto task = when_all(
+        // write chunk of data into one end repeatedly
+        pipe_write(wPipe, databuffer, scheduler, stopWrite.get_token()),
+        // read the data 1 byte at a time from the other end
+        [&, rPipe=std::ref(rPipe)]() -> unifex::task<void> {
+          co_await pipe_bench(rPipe, buffer, scheduler, WARMUP_DURATION, data, reps, offset);
+          offset = reps%sizeof(data);
+          reps = 0;
+          printf("warmup completed!\n");
+          // exclude the warmup time
+          start = end = std::chrono::high_resolution_clock::now();
+
+          co_await pipe_bench(rPipe, buffer, scheduler, BENCHMARK_DURATION, data, reps, offset);
+
+          end = std::chrono::high_resolution_clock::now();
+          printf("benchmark completed!\n");
+          auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
+              end - start)
+              .count();
+          auto ns = std::chrono::duration_cast<std::chrono::nanoseconds>(
+              end - start)
+              .count();
+          double reads = 1000000000.0 * reps / ns;
+          std::cout
+              << "completed in "
+              << ms << " ms, "
+              << ns << "ns, "
+              << reps << "ops\n";
+          std::cout
+              << "stats - "
+              << reads << "reads, "
+              << ns/reps << "ns-per-op, "
+              << reps/ms << "ops-per-ms\n";
+          stopWrite.request_stop();
+        }());
+#else
```

Comment:
This is an awful lot of code duplication, even if it's "just a test". Would you be willing to factor out the bits that are common?

Reply:
Yes, that was just to show usage with coroutines.

I rolled back *io_uring_test.cpp* and *epoll_test.cpp* and created a new *file_coroutines_test.cpp* to show usage.
However It requires io_uring test since *file_concepts* are not implemented neither for *epoll_context* or *low_latency_iocp_context*.